From edfd17b99d6579e575bbce5fe8a3b4691ca16bd7 Mon Sep 17 00:00:00 2001
From: "Luke D. Jones" <luke@ljones.dev>
Date: Sat, 17 Sep 2022 09:47:56 +1200
Subject: [PATCH 5/5] acpi/x86: s2idle: Add a quirk for ASUS ROG Zephyrus G14

Add a quirk for the ROG Zephyrus G14 GA402R<variant> laptop series
which has a similar issue to the FA707RE where the AMD s2idle
is missing an ASL method, and must use the Microsoft Modern Standby
calling path instead.

The DMI match needs to include the model number as "ROG Zephyrus G14"
includes previous generations of the laptop which don't have this issue.

Signed-off-by: Luke D. Jones <luke@ljones.dev>
---
 drivers/acpi/x86/s2idle.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/acpi/x86/s2idle.c b/drivers/acpi/x86/s2idle.c
index e2b73809ab50..0c8348de5cbc 100644
--- a/drivers/acpi/x86/s2idle.c
+++ b/drivers/acpi/x86/s2idle.c
@@ -420,6 +420,17 @@ static const struct dmi_system_id s2idle_dmi_table[] __initconst = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "ASUS TUF Gaming A17"),
 		},
 	},
+	{
+		/*
+		 * ASUS ROG Zephyrus G14 GA402R<variant> series
+		 * These laptops have a similar issue to the FA707RE
+		 */
+		.callback = lps0_prefer_microsoft,
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "ASUSTeK COMPUTER INC."),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ROG Zephyrus G14 GA402R"),
+		},
+	},
 	{}
 };
 
-- 
2.37.3

